utils/custom.js

```diff
@@ -113,23 +135,15 @@ const handleAnonymousData = async (
     childComment.map(async (child) => {
       const user = karmaScores.find((user) => user.user_id === child.user_id);
       if (child?.data?.anon_user_info_emoji_name) {
-        let childCommentLv2 = child?.latest_children?.comment || [];
-        childCommentLv2 = childCommentLv2?.map((child2) => {
-          const child2_user = karmaScores.find((user) => user.user_id === child2.user_id);
-          if (child2.data.anon_user_info_emoji_name) {
-            return {
-              ...child2,
-              user_id: null,
-              user: {},
-              target_feeds: [],
-              karmaScores: roundingKarmaScore(child2_user?.karma_score || 0)
-            };
-          }
-          return {...child2, karmaScores: roundingKarmaScore(child2_user?.karma_score || 0)};
-        });
+        let childCommentLv2Anon = setChildCommentLv2(
+          child?.latest_children?.comment,
+          karmaScores,
+          req.userId,
+          myAnonymousId
+        );
         return {
           ...child,
-          latest_children: {comment: childCommentLv2},
+          latest_children: {comment: childCommentLv2Anon},
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This section looks good. The `setChildCommentLv2` function is being used correctly to process level 2 comments for anonymous users. The resulting array is then correctly set as the `latest_children.comment` attribute of the parent comment.



<!-- This is an auto-generated comment by OSS CodeRabbit -->